python/benchmarks/basic_functions.py

```diff
@@ -257,62 +303,61 @@ def setup_cache(self):
-        self.df_append_single = generate_pseudo_random_dataframe(1, "s", get_time_at_fraction_of_df(1.1))
-        self.df_append_large = generate_pseudo_random_dataframe(rows, "s", get_time_at_fraction_of_df(2))
+        self.df_update_single = generate_pseudo_random_dataframe(1, "s", get_time_at_fraction_of_df(0.5, rows))
+        self.df_update_half = generate_pseudo_random_dataframe(rows//2, "s", get_time_at_fraction_of_df(0.75, rows))
+        self.df_update_upsert = generate_pseudo_random_dataframe(rows, "s", get_time_at_fraction_of_df(1.5, rows))
+        self.df_append_single = generate_pseudo_random_dataframe(1, "s", get_time_at_fraction_of_df(1.1, rows))
 
         self.df_update_short_wide = generate_random_floats_dataframe_with_index(
             ModificationFunctions.WIDE_DF_ROWS, ModificationFunctions.WIDE_DF_COLS
         )
-        self.df_append_short_wide = generate_random_floats_dataframe_with_index(
-            ModificationFunctions.WIDE_DF_ROWS, ModificationFunctions.WIDE_DF_COLS, "s", get_time_at_fraction_of_df(2, rows=ModificationFunctions.WIDE_DF_ROWS)
-        )
 
         self.ac = Arctic(ModificationFunctions.CONNECTION_STRING)
         self.lib = self.ac[get_prewritten_lib_name(rows)]
         self.lib_short_wide = self.ac[get_prewritten_lib_name(ModificationFunctions.WIDE_DF_ROWS)]
 
 
-    def teardown(self, rows):
+    def teardown(self, lad: LargeAppendDataModify, rows):
         # After the modification functions clean up the changes by replacing the modified ARCTIC_DIR with the original ARCTIC_DIR_ORIGINAL
         # TODO: We can use dirs_exist_ok=True on copytree instead of removing first if we run with python version >=3.8
         rmtree(ModificationFunctions.ARCTIC_DIR)
         copytree(ModificationFunctions.ARCTIC_DIR_ORIGINAL, ModificationFunctions.ARCTIC_DIR)
-        del self.ac
 
+        del self.ac
 
-    def time_update_single(self, rows):
+    def time_update_single(self, lad: LargeAppendDataModify, rows):
         self.lib.update(f"sym", self.df_update_single)
 
-    def time_update_half(self, rows):
+    def time_update_half(self, lad: LargeAppendDataModify, rows):
         self.lib.update(f"sym", self.df_update_half)
 
-    def time_update_upsert(self, rows):
+    def time_update_upsert(self, lad: LargeAppendDataModify, rows):
         self.lib.update(f"sym", self.df_update_upsert, upsert=True)
 
-    def time_update_short_wide(self, rows):
+    def time_update_short_wide(self, lad: LargeAppendDataModify, rows):
         self.lib_short_wide.update("short_wide_sym", self.df_update_short_wide)
 
-    def time_append_single(self, rows):
+    def time_append_single(self, lad: LargeAppendDataModify, rows):
         self.lib.append(f"sym", self.df_append_single)
 
-    def time_append_large(self, rows):
-        self.lib.append(f"sym", self.df_append_large)
+    def time_append_large(self, lad: LargeAppendDataModify, rows):
+        large: pd.DataFrame = lad.df_append_large[rows].pop()
```

Comment:
We are modifying the `lad` on each run of the benchmark. This relies on subsequent runs of the benchmarks accessing the modified `lad`. I'm not super confident that is the case.

In [here](https://asv.readthedocs.io/en/latest/using.html#environments) I see:
> To improve reproducibility, each benchmark is run in its own process.

Reply:
pop(0) changed. Also updated the header of PR with trace of failure

Reply:
fixed